Collector/MigrationsCollector.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MigrationsBundle\Collector;
+
+use Doctrine\Migrations\DependencyFactory;
+use Doctrine\Migrations\Metadata\AvailableMigration;
+use Doctrine\Migrations\Metadata\AvailableMigrationsList;
+use Doctrine\Migrations\Metadata\ExecutedMigration;
+use Doctrine\Migrations\Metadata\ExecutedMigrationsList;
+use Doctrine\Migrations\Metadata\Storage\TableMetadataStorageConfiguration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+class MigrationsCollector extends DataCollector
+{
+    /** @var DependencyFactory */
+    private $dependencyFactory;
+
+    public function __construct(DependencyFactory $dependencyFactory)
+    {
+        $this->dependencyFactory = $dependencyFactory;
+    }
+
+    public function collect(Request $request, Response $response, \Throwable $exception = null)
+    {
+        $metadataStorage = $this->dependencyFactory->getMetadataStorage();
+        $planCalculator = $this->dependencyFactory->getMigrationPlanCalculator();
+        $statusCalculator = $this->dependencyFactory->getMigrationStatusCalculator();
+
+        $availableMigrations = $planCalculator->getMigrations();
```

Comment:
What is the performance cost of collecting all this information about migrations ? To me, this will bring value only if the status is not green (in which case there is a good chance chance that you get an exception page because the ORM tries to query non-existent fields or tables).
If this takes time to collect, the cost/benefit ratio might not be good enough for DX.

Reply:
reduced the profiler data to the names only, loading data using a twig extension.
should we add an config key to disable the profiler, like the original PR proposed?

Reply:
@stof are you referring to the cost of getting the data from doctrine migrations or the size on disk of the collected profile?